torch/csrc/jit/script/compiler.cpp

```diff
@@ -231,36 +224,41 @@ struct Environment {
   ValueTable value_table;
 };
 
-Const getAttributeValue(Expr value_expr) {
-  switch (value_expr.kind()) {
-    case TK_CONST: {
-      return Const(value_expr);
-    } break;
-    case TK_TRUE: {
-      return Const::create(value_expr.range(), "1");
-    } break;
-    case TK_FALSE: {
-      return Const::create(value_expr.range(), "0");
-    } break;
-    default:
-      throw ErrorReport(value_expr) << "attributes must be constants, or a list of constants";
-      break;
-  }
-}
-
 std::shared_ptr<SugaredValue> packOutputs(Graph& g, at::ArrayRef<Value*> values) {
   if(values.size() == 1) {
     return std::make_shared<SimpleValue>(values[0]);
   }
   return std::make_shared<SimpleValue>(g.insertNode(g.createTuple(values))->output());
 }
 
+at::Tensor getConstantValue(const SourceRange& loc, Value* v, int64_t max_dim) {
+  if(v->node()->kind() == prim::Constant) {
+    auto t = v->node()->t(attr::value);
+    if(t.ndimension() > max_dim) {
+      throw ErrorReport(loc) << "attributes must be scalars or lists of scalars";
```

Comment:
Passing in `max_dim` and hardcoding an error message for scalars doesn't seem like a good idea. Just put the `0` in the code.

Reply:
What zero? In one call max_dim is 1 and in the other it is 0

Reply:
Yeah, but I'm not sure why would we want it to be `1` in the other case. We never allow passing in tensors via kwargs, they're supposed to be scalars or lists of scalars from what I understand.

Reply:
I think I see what you are saying, let me see if that breaks anything we need.

Furthermore, I've changed my mind and think we _should_ add `float` and `int` to the script to mirror python and type check the code to ensure it is used in a way that would be a legit subset of the pytorch program. Doing otherwise makes it hard to ensure the "subset" property of @script. We can and should do this in a way that doesn't change the _IR_ and only change the typechecker/compiler.cpp.   Once we have verified that type-correctness of the @script program, we can convert float/int values into Tensors in the IR for the purposes of execution. This keeps this decision about the type system from affecting later things.